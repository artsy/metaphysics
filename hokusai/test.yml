version: "2"
services:
  metaphysics:
    command: env NODE_OPTIONS=--max_old_space_size=4096 yarn ci
    extends:
      file: build.yml
      service: metaphysics
    environment:
      - MEMCACHED_URL=metaphysics-memcached:11211
      - CI_PULL_REQUEST=$CI_PULL_REQUEST
      - CIRCLE_PULL_REQUEST=$CIRCLE_PULL_REQUEST
      - CIRCLE_BUILD_NUM=$CIRCLE_BUILD_NUM
      - ENABLE_CONSIGNMENTS_STITCHING=true
    env_file: ../.env.test
    depends_on:
      - metaphysics-memcached
  metaphysics-memcached:
    image: memcached
    ports:
      - 11211:11211
