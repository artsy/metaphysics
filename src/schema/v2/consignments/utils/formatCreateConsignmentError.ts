import {
  GraphQLInt,
  GraphQLNonNull,
  GraphQLObjectType,
  GraphQLString,
} from "graphql"
import { formatGravityError } from "lib/gravityErrorHandler"
import { ResolverContext } from "types/graphql"

export const ConsignmentInquiryMutationErrorName =
  "ConsignmentInquiryMutationError"

export const ConsignmentInquiryErrorType = new GraphQLObjectType<
  any,
  ResolverContext
>({
  name: ConsignmentInquiryMutationErrorName,
  fields: () => ({
    type: {
      type: GraphQLString,
    },
    message: {
      type: new GraphQLNonNull(GraphQLString),
    },
    error: {
      type: GraphQLString,
    },
    statusCode: {
      type: GraphQLInt,
    },
  }),
})

export const formatCreateConsignmentError = (error) => {
  return formatGravityError(error)
}
