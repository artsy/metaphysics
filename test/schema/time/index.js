import { runQuery } from "test/utils"

describe("Time type", () => {
  it("returns the system time", () => {
    const fullQuery = `
    {
      time {
        day
        wday
        month
        year
        hour
        min
        sec
        dst
        unix
        utc_offset
        zone
        iso8601
      }
    }
    `
    const rootValue = {
      systemLoader: () =>
        Promise.resolve({
          day: 15,
          wday: 3,
          month: 11,
          year: 2017,
          hour: 19,
          min: 45,
          sec: 3,
          dst: false,
          unix: 1510775103,
          utc_offset: 0,
          zone: "UTC",
          iso8601: "2017-11-15T19:45:03Z",
        }),
    }

    expect.assertions(1)
    return runQuery(fullQuery, rootValue).then(response => {
      expect(response).toMatchSnapshot()
    })
  })
})
