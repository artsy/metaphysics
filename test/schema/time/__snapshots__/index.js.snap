// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Time type returns the system time 1`] = `
Object {
  "time": Object {
    "day": 15,
    "dst": false,
    "hour": 19,
    "iso8601": "2017-11-15T19:45:03Z",
    "min": 45,
    "month": 11,
    "sec": 3,
    "unix": 1510775103,
    "utc_offset": 0,
    "wday": 3,
    "year": 2017,
    "zone": "UTC",
  },
}
`;
